core/src/main/scala/org/apache/spark/deploy/security/README.md

```diff
@@ -0,0 +1,238 @@
+# Delegation Token Handling In Spark
+
+This document aims to explain and demystify delegation tokens as they are used by Spark, since
+this topic is generally a huge source of confusion.
+
+
+## What are delegation tokens?
+
+Delegation tokens (DTs from now on) are authentication tokens used by some services to replace
+Kerberos service tokens. Many services in the Hadoop ecosystem have support for DTs, since they
+have two very desirable advantages over Kerberos tokens:
+
+* No need to distribute Kerberos credentials
+
+In a distributed application, distributing Kerberos credentials is tricky. Not all users have
+keytabs, and when they do, it's generally frowned upon to distribute them over the network as
+part of application data.
```

Comment:
We seem to be mixing the section of what is a delegation token with what the benefits are, the definition of what also seems to be in  the next section with the lifecycle, perhaps we need to reorganize this a bit to have more of a section on what is DT, within there you can go into the benefits or why

Might mention here that with tokens you can't acquire more tokens vs with tgts or keytab you can get more tokens so its more secure to use tokens.